utils/utils.go

```diff
@@ -21,7 +22,11 @@ func DecodeYAMLFile(file *os.File) (*ContextMap, error) {
 }
 
 func WriteYamlFile(file *os.File, values *ContextMap) error {
-	return yaml.NewEncoder(file).Encode(&values)
+	b, err := yaml.Marshal(values)
+	if err != nil {
+		return err
+	}
+	return ioutil.WriteFile(file.Name(), b, readWritePerms)
```

Comment:
Or perhaps better, just pass the filename in the argument

Reply:
Do you mean?
```go
func WriteYamlFile(filename string, values *ContextMap) error {
	b, err := yaml.Marshal(values)
	if err != nil {
		return err
	}
	return ioutil.WriteFile(filename, b, readWritePerms)
}
```

Reply:
Strong suggestion rather than requirement but I suggest you move utils/utils.go into config/yaml.go and config/homedir.go 

consolidating unrelated code into the same shared compilation unit or module (e.g. creating "StringUtils.java" ) or "common.go" is usually a bad idea in any software - (everybody does it/ has done it  -they are all wrong and need to get in the see) .

There are a couple of fundamental problems: 
* The code in these modules has low coheseion (lots of independent bits of functionality that don't use or relate to one another) 
* The next person sees "utils" or "common" and pehaps sees it a different way and then decides to dump their random bitty code in there  - after a while you end up with a huge ball of unrelated code

Reply:
Ok sure, that makes a lot more sense, especially as a lot of the calls were made from inside the config/ package anyway!